### PR TITLE
Swift 4.0 support

### PR DESCRIPTION
**Pull request by rynecheow**

### CODE REVIEW

File: Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ import RxCocoa
 public enum UITapGestureRecognizerDefaults {
     public static var numberOfTouchesRequired: Int = 1
     public static var numberOfTapsRequired: Int = 1
-    public static var configuration: ((UITapGestureRecognizer, RxGestureRecognizerDelegate) -> Void)? = nil
+    public static var configuration: ((UITapGestureRecognizer, RxGestureRecognizerDelegate) -> Void)?
```
Comment (jegnux) [inline Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift:29]: Why did you remove this default value? Does it prevent compilation or it was just enforced by a warning (maybe from swiftlint)?

File: Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift [hunk 2]
```diff
@@ -43,10 +43,10 @@ public struct TapGestureRecognizerFactory: GestureRecognizerFactory {
      - parameter configuration: A closure that allows to fully configure the gesture recognizer
      */
     public init(
-        numberOfTouchesRequired: Int = Defaults.numberOfTouchesRequired,
-        numberOfTapsRequired: Int = Defaults.numberOfTapsRequired,
-        configuration: ((UITapGestureRecognizer, RxGestureRecognizerDelegate) -> Void)? = Defaults.configuration
-        ){
+        numberOfTouchesRequired: Int = UITapGestureRecognizerDefaults.numberOfTouchesRequired,
+        numberOfTapsRequired: Int = UITapGestureRecognizerDefaults.numberOfTapsRequired,
+        configuration: ((UITapGestureRecognizer, RxGestureRecognizerDelegate) -> Void)? = UITapGestureRecognizerDefaults.configuration
```
Comment (jegnux) [inline Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift]: Why did you replace `Default` by the whole name? The typealias trick was quite good.

**Review by jegnux — COMMENTED** [review]
Hi! thanks for you contribution.
You make me realize that we forgot macOS on swift4 branch. Glad you restore some fairness :)

File: Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift (hunk 1 above)
Comment (rynecheow) [reply Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift:29]: You are right on because of the warning on Swiftlint. Is it an intention to make it explicitly `nil` as Optional types are `nil` implicitly by default. If so I could re-add them back.

File: Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift (hunk 2 above)
Comment (rynecheow) [reply Pod/Classes/iOS/UITapGestureRecognizer+RxGesture.swift]: I'll change this. I did not realise this part honestly.

**Review by jegnux — APPROVED** [review]

**Comment by jegnux** [comment]
@rynecheow Great ! Thanks for you contribution 🙏

**Comment by rynecheow** [comment]
Can we merge this?
